ambari-server/src/main/java/org/apache/ambari/server/serveraction/kerberos/KerberosServerAction.java

```diff
@@ -452,10 +452,13 @@ protected CommandReport processIdentities(Map<String, Object> requestSharedDataC
         if (serviceComponentFilter != null && pruneServiceFilter()) {
           kerberosKeytabController.adjustServiceComponentFilter(clusters.getCluster(getClusterName()), true, serviceComponentFilter);
         }
-        final Collection<KerberosIdentityDescriptor> serviceIdentities = serviceComponentFilter == null ? null : kerberosKeytabController.getServiceIdentities(getClusterName(), serviceComponentFilter.keySet());
-        for (ResolvedKerberosKeytab rkk : kerberosKeytabController.getFilteredKeytabs(serviceIdentities, getHostFilter(),getIdentityFilter())) {
+        final Collection<KerberosIdentityDescriptor> identities = serviceComponentFilter == null ? null : kerberosKeytabController.getServiceIdentities(getClusterName(), serviceComponentFilter.keySet());
+        if (identities != null && getOperationType(getCommandParameters()) == OperationType.RECREATE_ALL){
+          identities.addAll(kerberosHelper.getGlobalActiveIdentities(getClusterName()));
```

Comment:
If `getGlobalActiveIdentities` is a superset of the identities returned by `getServiceIdentities` then why do both.  Wouldn't it be better to do something lie
```
if (getOperationType(getCommandParameters()) == OperationType.RECREATE_ALL) {
  identities = kerberosHelper.getGlobalActiveIdentities(getClusterName());
}
else {
    identities = serviceComponentFilter == null ? null : kerberosKeytabController.getServiceIdentities(getClusterName(), serviceComponentFilter.keySet())
}
```

Reply:
`getGlobalActiveIdentities` is not a superset. It just contains `smokeuser` and `spnego`. The `identities` branch that contains them is on the same level of hierarchy with `services` branch. If the name is misleading, maybe there is a proper name for them?

Reply:
Thanks for the clarification. Can you update the Javadoc for `getGlobalActiveIdentities` to indicate that?  Naming is hard, so I am not sure if there is a better name, now that I know what the method is supposed to do.